[MCH] added DPL message with orbits information

aferrero2707 opened the pull request:
The additional message provides the list of processed orbits for each CRU link in the input raw buffer.
Useful for example to estimate the detector occupancy in continuous readout mode.

aferrero2707 committed:
[MCH] added DPL message with orbits information

The additional message provides the list of processed orbits for each CRU link in the input raw buffer.
Useful for example to estimate the detector occupancy in continuous readout mode.

aphecetche commented on Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:
```diff
@@ -149,12 +149,25 @@ class DataDecoderTask
         o2::raw::RDHUtils::printRDH(rdhAny);
       }
+
+      uint64_t orbitInfo = orbit;
```
I have a few comments about the orbitInfo treatment : 

- I assume the encoding of (orbit, linkid, feeid) won't be used only in this file, so it should be extracted to its own function I guess (with a corresponding decoding one most probably ? )
- the creation of the orbit info should not be placed in the patchPage function which is supposed to disappear at some point in the future (while the orbitInfo will not, right ?)
- why not using a std::set instead of the vector here ? (would avoid the find)

aferrero2707 replied:
@aphecetche I have created a separate `OrbitInfo` class with getter functions for orbit/linkID/feeID fields, and moved the creation out of the patch page function.

Concerning the use of `std::set`, the problem is that it does not provide a `data()` function to access the underlying storage buffer, so one would need to copy elements one-by-one instead of using a `memcpy()`. I have therefore kept the original `std::vector`.

aphecetche commented on Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:
```diff
@@ -297,23 +312,33 @@ class DataDecoderTask
-    char* outbuffer = nullptr;
-    outbuffer = (char*)realloc(outbuffer, OUT_SIZE);
-    memcpy(outbuffer, digits.data(), OUT_SIZE);
+    const size_t digitsSize = sizeof(o2::mch::Digit) * digits.size();
```
the two pieces of code below are just repetitions, could be extracted to a (templatized) function, e.g. `char* createBuffer(T container)` ?

aferrero2707 replied:
Done.

aferrero2707 committed:
[MCH] introduced OrbitInfo class

aferrero2707 committed:
[MCH] added orbit info printout

aferrero2707 committed:
[MCH] applied clang code formatting

aferrero2707 committed:
[MCH] using std::unordered_set to store the list of processed orbits

The source files fot the OrbitInfo class have also been moved
inside the MCH raw decoder folder.

aferrero2707 committed:
[MCH] fixed clang formatting
